### DISCUSSION TITLE
tutorials/intro-typescript/12-adding-the-amenity-type

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#184] [course feedback]
# tutorials/intro-typescript/12-adding-the-amenity-type

https://www.apollographql.com/tutorials/intro-typescript/12-adding-the-amenity-type

**Comment by kelvenus** [comment]
We got the error because amenity is an Array, so we sure be returning an array. I do not know how to return the array values, so I will go to the next section.

**Reply by mabuyo** [reply]
Hey there! To clarify, we get the error because the `/featured-listings` endpoint returns a response that only includes listing id values. Name and category aren't fields that are included in the response. It's already in array form. Hope that helps! :)

**Reply by kelvenus** [reply]
Thanks for your reply.

**Comment by Ressvieth** [comment]
Hey, just a small inconsistency - in the first examples on this page we have non-nullable values:
```
  numOfBeds: Int!
  costPerNight: Float!
```
while in the final code for the `schema.graphql` we have:
``` 
  "The number of beds available"
  numOfBeds: Int
  "The cost per night"
  costPerNight: Float
